packages/devextreme/testing/tests/DevExpress.ui.widgets/dropDownButton.tests.js

```diff
@@ -1565,19 +1564,16 @@ QUnit.module('common use cases', {
             useSelectMode: true,
             selectedItemKey: 1
         });
-
         const items = [{ id: 3, name: 'test' }];
         this.dropDownButton.option('items', items);
 
-        setTimeout(() => {
-            assert.strictEqual(this.dropDownButton.option('selectedItemKey'), null, 'selectedItemKey is correct');
-            assert.strictEqual(this.dropDownButton.option('selectedItem'), null, 'selectedItem is correct');
+        assert.strictEqual(this.dropDownButton.option('selectedItemKey'), null, 'selectedItemKey is correct');
+        assert.strictEqual(this.dropDownButton.option('selectedItem'), null, 'selectedItem is correct');
 
-            const list = getList(this.dropDownButton);
-            assert.deepEqual(list.option('selectedItemKeys'), [], 'list selectedItemKey is kept');
-            assert.strictEqual(list.option('selectedItem'), undefined, 'list selectedItem is correct');
-            done();
-        });
+        const list = getList(this.dropDownButton);
+        assert.deepEqual(list.option('selectedItemKeys'), [], 'list selectedItemKey is kept');
+        // TODO research me
```

Comment:
I don't see what should happen to force us to research it later. I think we should deal with it now.
Let me check a bit later.

Reply:
I added a `console.log` to List's option change and found the following difference:
before:
```text
OPTIONS CHANGE selectedItem undefined  ->  null
OPTIONS CHANGE selectedItem {id: 1, name: 'a'}  ->  null
OPTIONS CHANGE selectedItem {id: 1, file: 'vs.exe', name: 'Trial for Visual Studio', icon: 'box'}  ->  null
```
after:
```text
OPTIONS CHANGE selectedItem undefined  ->  null
OPTIONS CHANGE selectedItem {id: 1, name: 'a'}  ->  null
OPTIONS CHANGE selectedItem undefined  ->  null
```
U can see that 3rd change is made with different `previousValue`. 

**Why this difference is a reason of a red test?**

All complexity of this behavior is related to selection options sync in collection: [link](https://github.com/DevExpress/DevExtreme/blob/24_1/packages/devextreme/js/ui/collection/ui.collection_widget.edit.js#L299). If we change `selectedItem` option from one nullish value to another (for example, from undefined to null), no addtiional sync is made.

---

**Why does the last `selectedItem` option change in the last version have such a strange previous value?**

Because of this [call](https://github.com/DevExpress/DevExtreme/blob/24_1/packages/devextreme/js/ui/drop_down_button.js#L629).
The `_loadSelectedItem` promise is resolved with `selectedItem=null` but we anyway change `selectedItemKeys` to `[1]` and it leads to update of `selectedItem` on options sync in collection.

--- 

**Why does the last `selectedItem` option change in the updated version have an `undefined` as a previous value if we still have a call mentioned in the previous question?**

Because when we try to sync options dataSource [loads](https://github.com/DevExpress/DevExtreme/blob/24_1/packages/devextreme/js/ui/selection/selection.strategy.standard.js#L227) nothing in the new version, but loads an old item in the previous version. **Why? I suggest u debug this.**

Actual issue can be resolved if we change `resolve` with a `reject` [here](https://github.com/DevExpress/DevExtreme/blob/24_1/packages/devextreme/js/ui/drop_down_button.js#L215). It seems to be more correct from idealogical point of view.